test/testfixture/query_functions.go

```diff
@@ -0,0 +1,62 @@
+package testfixture
+
+import (
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/iteration"
+	"github.com/fabric8-services/fabric8-wit/label"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+	uuid "github.com/satori/go.uuid"
+)
+
+// LabelByName returns the first label that has the given name (if any). If you
+// have labels with the same name in different spaces you can also pass in one
+// space ID to filter by space as well.
+func (fxt *TestFixture) LabelByName(name string, spaceID ...uuid.UUID) *label.Label {
```

Comment:
Why accepting a slice of spaceIDs? I did not understand the use case. Same with all following query_funcitons. Can you please elaborate? And I see it is not being used other than for counting.

Reply:
Suppose you create two spaces and inside of those space you create a label "foo". Then you will end up with a test fixture that contains two labels in the `fxt.Labels` slice. Both have the name "foo" so you need to distinguish the two by their space ID. The ellipsis argument is only there to make it optional.

Reply:
Thanks for explanation @kwk 
> The ellipsis argument is only there to make it optional.

This was the crux, got it!